src/Symfony/Bundle/FrameworkBundle/Command/FormDebugCommand.php

```diff
@@ -0,0 +1,233 @@
+            $io->section('Type extensions');
+            $io->listing(array_keys($this->extensions));
+        }
+    }
+
+    private function getOptionsGroup(ResolvedFormTypeInterface $type)
+    {
+        $innerOptionsResolver = new OptionsResolver();
+        try {
+            $type->getInnerType()->configureOptions($innerOptionsResolver);
+        } catch (UndefinedOptionsException $e) {
+        }
+
+        if (null !== $type->getParent()) {
+            $this->optionsResolver = $this->getParentOptionResolver($type->getParent());
+        } else {
+            $this->optionsResolver = new OptionsResolver();
+        }
+
+        $options['inner'] = array_diff($innerOptionsResolver->getDefinedOptions(), $this->optionsResolver->getDefinedOptions());
+        $options['overridden'] = array_intersect($innerOptionsResolver->getDefinedOptions(), $this->optionsResolver->getDefinedOptions());
+
+        $options['inherited'] = array();
+        foreach ($this->parents as $class => $opts) {
+            $options['inherited'][$class] = array_diff($opts, $options['overridden']);
+        }
+
+        $type->getInnerType()->configureOptions($this->optionsResolver);
+        $inheritedOptions = $this->optionsResolver->getDefinedOptions();
+        foreach ($type->getTypeExtensions() as $extension) {
+            $extension->configureOptions($this->optionsResolver);
+            $this->extensions[get_class($extension)] = array_diff($this->optionsResolver->getDefinedOptions(), $inheritedOptions);
+        }
+        $options['extension'] = array();
+        foreach ($this->extensions as $class => $opts) {
+            $options['extension'][$class] = array_diff($opts, $options['overridden']);
+        }
+
+        return array_filter($options);
+    }
+
+    private function getParentOptionResolver(ResolvedFormTypeInterface $type)
+    {
+        $parentClass = get_class($type->getInnerType());
+        $this->parents[$parentClass] = array();
+
+        if (null !== $type->getParent()) {
+            $optionsResolver = clone $this->getParentOptionResolver($type->getParent());
+        } else {
+            $optionsResolver = new OptionsResolver();
+        }
+
+        $inheritedOptions = $optionsResolver->getDefinedOptions();
+        $type->getInnerType()->configureOptions($optionsResolver);
+        $this->parents[$parentClass] = array_diff($optionsResolver->getDefinedOptions(), $inheritedOptions);
+
+        foreach ($type->getTypeExtensions() as $extension) {
+            $inheritedOptions = $optionsResolver->getDefinedOptions();
+            $extension->configureOptions($optionsResolver);
+            $this->extensions[get_class($extension)] = array_diff($optionsResolver->getDefinedOptions(), $inheritedOptions);
+        }
+
+        return $optionsResolver;
+    }
+
+    private function getBuiltinFormTypeClass($class)
+    {
+        $namespaces = array(
+            (new \ReflectionClass(FormType::class))->getNamespaceName(),
```

Comment:
No need to use reflection here to me. It'll also fail in case the doctrine bridge is not used for instance (that's an issue if we want this command to be in the Form component, or simply if you don't use doctrine in your symfony app)